test/src/error-snippets/frames.ts

```diff
@@ -0,0 +1,141 @@
+                repl: "    at REPL1:1:1",
+                internal1: "    at Module._compile (node:internal/modules/cjs/loader:1159:14)",
+                internal2: "    at Script.runInThisContext (node:vm:129:12)",
+            },
+        } as const;
+
+        let findRelevantStackFrame: (err: Error) => SufficientStackFrame | null;
+
+        const buildError_ = (...errFrames: string[]): Error => {
+            if (!errFrames.length) {
+                errFrames = [frames.browser.wdio, frames.node.wdio];
+            }
+
+            const error = new Error("foo");
+
+            error.stack = `${error.name}: ${error.message}\n${errFrames.join("\n")}`;
+
+            return error;
+        };
+
+        beforeEach(() => {
+            logger = { warn: sandbox.stub() };
+            findRelevantStackFrame = proxyquire("../../../src/error-snippets/frames", {
+                "../utils/logger": logger,
+            }).findRelevantStackFrame;
+        });
+
+        it("should return null without throwing exceptions on bad input", () => {
+            const stackFrame = findRelevantStackFrame("foo" as unknown as Error);
+
+            assert.calledOnceWith(logger.warn, "Unable to find relevant stack frame:");
+            assert.isNull(stackFrame);
+        });
+
+        it("should not return repl frames", () => {
+            const error = buildError_(frames.browser.repl, frames.node.repl);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            console.log(stackFrame);
+            assert.notCalled(logger.warn);
+            assert.isNull(stackFrame);
+        });
+
+        it("should not return internal frames", () => {
+            const error = buildError_(frames.node.internal1, frames.node.internal2);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.notCalled(logger.warn);
+            assert.isNull(stackFrame);
+        });
+
+        it("should return wdio frame, if there is nothing left", () => {
+            const error = buildError_(frames.node.internal1, frames.node.repl, frames.browser.wdio);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "async Element.elementErrorHandlerCallbackFn");
+        });
+
+        it("should return first seen frame with equal relevance", () => {
+            const error = buildError_(frames.node.internal2, frames.node.repl, frames.node.wdio, frames.browser.wdio);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "Element.elementErrorHandlerCallbackFn");
+        });
+
+        it("should value other project deps frame more, than wdio frame", () => {
+            const error = buildError_(frames.browser.wdio, frames.browser.unknownWdio, frames.node.projectDeps);
+
+            const stackFrame = findRelevantStackFrame(error);
+
+            assert.equal(stackFrame?.functionName, "Object.tryCatcher");
```

Comment:
it would be better to pass the value being checked directly in the test. now it is difficult to understand what this value is